tableaudocumentapi/datasource.py

```diff
@@ -59,6 +59,31 @@ def from_file(cls, filename):
         dsxml = ET.parse(filename).getroot()
         return cls(dsxml, filename)
 
+    def get_connections_by_server_name(self, server_name):
+        matches = []
+        if server_name:
+            for connection in self._connections:
+                if str(server_name) == connection.server:
+                    matches.append(connection)
+
+        return matches
+
+    def get_connections_by_attributes(self, conn_attributes):
+        from itertools import product
+
+        matches = []
+
+        valid_attributes = ('server', 'username', 'dbclass', 'dbname')
+
+        if not (all(attr in valid_attributes for attr in conn_attributes) and conn_attributes):
+            raise RuntimeError("That is not a valid connection attribute")
+
+        for attr, conn in product(conn_attributes, self._connections):
+            if getattr(conn, attr).lower() == conn_attributes[attr].lower():
+                matches.append(conn)
+
+        return matches
+
```

Comment:
I vote samples. We build a solid, focused core api with some great samples. We can evaluate as we get more mature what we might add to make really common cases better but we are far from that yet and this is not "core". THanks!